### DISCUSSION TITLE
How do we keep main and next in sync?

### DISCUSSION THREAD

**Discussion by ronag** [nodejs/undici#2849] [General]
What is the workflow for this? In my private projects I usually just rebase and force push this kind of "next gen" branch. But I don't think that works here.

**Comment by ronag** [comment]
@mcollina

**Comment by Uzlopak** [comment]
I feel you. Maybe @climba03003 knows

**Comment by climba03003** [comment]
`force push with rebase` will be crediting you as an author, it lost all the unmodified commit.
So, it should be use for `Pull Request` or `Backporting` only, similar to `cherry pick`.

In order to keep the original history, you should instead using `no fast forward` which is `--no-ff` when merging to `next`.
You may notice there will be conflict at this stage.

The world would be easier to follow how `nodejs/node`.
The `main` is actually the `next` branch and always `cherrypick` to the version branch.
In this way, the conflict will be much less and of-cause it increase the workload of publisher.

**Reply by ronag** [reply]
> The main is actually the next branch and always cherrypick to the version branch.
> In this way, the conflict will be much less and of-cause it increase the workload of publisher.

I kind of like this... @mcollina?
